File: server/src/main/java/com/fluffy/exam/ui/request/UpdateExamQuestionsWebRequest.java
package com.fluffy.exam.ui.request;

import com.fluffy.exam.application.request.UpdateExamQuestionsRequest;
import com.fluffy.exam.application.request.question.QuestionRequest;
import com.fluffy.global.web.Accessor;
import java.util.List;

public record UpdateExamQuestionsWebRequest(List<QuestionRequest> questions) {

    public UpdateExamQuestionsRequest toAppRequest(Long examId, Accessor accessor) {
        return new UpdateExamQuestionsRequest(examId, questions, accessor);
    }
}

File: server/src/main/kotlin/com/fluffy/exam/ui/request/UpdateExamQuestionsWebRequest.kt
package com.fluffy.exam.ui.request

import com.fluffy.exam.application.request.UpdateExamQuestionsRequest
import com.fluffy.exam.application.request.question.QuestionRequest
import com.fluffy.global.web.Accessor

data class UpdateExamQuestionsWebRequest(val questions: List<QuestionRequest>) {

    fun toAppRequest(examId: Long, accessor: Accessor): UpdateExamQuestionsRequest {
        return UpdateExamQuestionsRequest(
            examId = examId,
            questions = questions,
            accessor = accessor
        )
    }
}
